src/anndata/_core/aligned_mapping.py

```diff
@@ -272,20 +278,23 @@ def dim_names(self) -> pd.Index:
         return (self.parent.obs_names, self.parent.var_names)[self._axis]
 
 
+# TODO: vals can't be None
 class AxisArrays(AlignedActualMixin, AxisArraysBase):
     def __init__(
         self,
         parent: AnnData | Raw,
         axis: int,
-        vals: Mapping | AxisArraysBase | None = None,
+        vals: dict[str, V] | AxisArraysBase | None = None,
     ):
         self._parent = parent
         if axis not in (0, 1):
             raise ValueError()
         self._axis = axis
-        self._data = dict()
-        if vals is not None:
-            self.update(vals)
+        if vals is None:
+            vals = {}
```

Comment:
The codecov is real but likely was there before since the logic did not change (in theory)